web/styles/message_row.css

```diff
@@ -474,6 +159,317 @@
+        text-align: end;
+        grid-area: time;
+    }
+
+    .message_content {
+        grid-area: message;
+        /*
+        Space between two single line messages is kept in
+        lockstep with adjacent Markdown elements.
+        */
+        padding: var(--message-box-markdown-aligned-vertical-space) 0 0;
+        color: var(--color-text-message-default);
+        font-size: var(--base-font-size-px);
+        line-height: var(--base-line-height-unitless);
+        min-height: 17px;
+        display: block;
+        position: relative;
+        overflow: hidden;
+
+        &:empty {
+            display: none;
+        }
+
+        &.condensed {
+            max-height: 8.5em;
+            min-height: 0;
+            overflow: hidden;
+            mask-image: linear-gradient(
+                to top,
+                hsl(0deg 0% 100% / 0%) 0%,
+                hsl(0deg 0% 100%) 60px
+            );
+            mask-size: cover;
+        }
+
+        &.collapsed {
+            padding: 0;
+            max-height: 0;
+            min-height: 0;
+            overflow: hidden;
+        }
+    }
+
+    .message_reactions {
+        grid-area: reactions;
+        display: flex;
+        /* Allow reactions to wrap in mobile viewports */
+        flex-wrap: wrap;
+        /* Keep reactions aligned to start (top), so that the
+           margin on the reaction button doesn't appear to
+           stretch the reactions taller. */
+        align-items: flex-start;
+        /* Control reaction spacing with a gap */
+        gap: 4px;
+    }
+
+    .message_edit {
+        grid-area: message;
+        /* Align self to start, rather than baseline, so the baseline
+          is preserved from the message itself--keeping the time,
+          edit message, and controls at the same vertical alignment. */
+        align-self: start;
+        margin-top: var(--message-box-vertical-margin);
+    }
+
+    .message_length_controller {
+        grid-area: more;
+    }
+
+    .collapsed ~ .message_length_controller {
+        /* When content is collapsed, the length controller
+           should occupy the space ordinarily assigned to
+           the message. */
+        grid-area: message;
+        /* However, don't let the SHOW MORE button participate
+           in the baseline group. */
+        align-self: start;
+
+        .message_length_toggle {
+            /* Ensure that a collapsed message maintains the
+               same space around the toggle button as any other
+               message-row content. */
+            margin: 4px 0;
+        }
+    }
+
+    &.content_edit_mode .message_length_controller {
+        /* If entering edit mode on a collapsed message,
+           just hide the controller area. This preserves
+           the collapsed state of the message, which need
+           not be touched. We just need to hide the button.
+           This works for condensed messages, too. */
+        display: none;
+    }
+}
+
+.include-sender {
```

Comment:
I think for this one, I liked the more specific `.messagebox.include-sender` selector -- `include-sender` alone feels too generic to be a top-level selector.

(We could also rename it to `messagebox-include-sender` or something if that's preferred)

Reply:
This is now addressed in #29938